src/main/java/org/folio/circulation/domain/policy/lostitem/LostItemPolicy.java

```diff
@@ -1,51 +1,73 @@
 import static org.folio.circulation.support.JsonPropertyFetcher.getBooleanProperty;
 import static org.folio.circulation.support.JsonPropertyFetcher.getObjectProperty;
 import static org.folio.circulation.support.JsonPropertyFetcher.getProperty;
 
 import java.math.BigDecimal;
+import java.util.Optional;
 
 import org.folio.circulation.domain.policy.Policy;
 
 import io.vertx.core.json.JsonObject;
 
 public class LostItemPolicy extends Policy {
-  private final ChargeAmount chargeAmountItem;
+  private final ItemFee setCostChargeFee;
   private final boolean chargeAmountItemPatron;
-  private final BigDecimal lostItemProcessingFee;
+  private final ItemFee processingChargeFee;
 
-  private LostItemPolicy(String id, String name, ChargeAmount chargeAmountItem,
-    boolean chargeAmountItemPatron, BigDecimal lostItemProcessingFee) {
+  private LostItemPolicy(String id, String name, ItemFee setCostChargeFee,
+    boolean chargeAmountItemPatron, ItemFee processingChargeFee) {
 
     super(id, name);
-    this.chargeAmountItem = chargeAmountItem;
+    this.setCostChargeFee = setCostChargeFee;
     this.chargeAmountItemPatron = chargeAmountItemPatron;
-    this.lostItemProcessingFee = lostItemProcessingFee;
+    this.processingChargeFee = processingChargeFee;
   }
 
   public static LostItemPolicy from(JsonObject lostItemPolicy) {
     return new LostItemPolicy(
       getProperty(lostItemPolicy, "id"),
       getProperty(lostItemPolicy, "name"),
-      ChargeAmount.from(getObjectProperty(lostItemPolicy, "chargeAmountItem")),
+      getSetCostFee(lostItemPolicy),
       getBooleanProperty(lostItemPolicy, "chargeAmountItemPatron"),
-      getBigDecimalProperty(lostItemPolicy, "lostItemProcessingFee")
+      getProcessingFee(lostItemPolicy)
     );
   }
 
-  public ChargeAmount getChargeAmountItem() {
-    return chargeAmountItem;
+  private static ItemFee getProcessingFee(JsonObject policy) {
+    final BigDecimal amount = getBigDecimalProperty(policy, "lostItemProcessingFee");
+    return amount != null
+      ? new ItemFee(amount)
+      : null;
+  }
+
+  private static ItemFee getSetCostFee(JsonObject policy) {
+    final JsonObject chargeAmountItem = getObjectProperty(policy, "chargeAmountItem");
+    final ChargeAmountType chargeType = forValue(getProperty(chargeAmountItem, "chargeType"));
+
+    if (chargeAmountItem == null || chargeType != SET_COST) {
+      return null;
+    }
+
+    return new ItemFee(getBigDecimalProperty(chargeAmountItem, "amount"));
+  }
+
+  public Optional<ItemFee> getSetCostChargeFee() {
+    return ofNullable(setCostChargeFee);
   }
 
   public boolean shouldChargeProcessingFee() {
     // Renamed to better describe intent. Fee/fines domain name quite unclear.
     return chargeAmountItemPatron;
   }
 
-  public BigDecimal getLostItemProcessingFee() {
-    return lostItemProcessingFee;
+  public Optional<ItemFee> getLostItemProcessingFee() {
```

Comment:
Why does this need to return an optional value?

Could there instead be a special kind of item fee where nothing happens?

Reply:
I need a way to understand whether the a cost was not defined at all or it has a zero value. It will be tricky with a wrapper.

Reply:
Why might it be tricky with a wrapper?

As I understand it, the item fee needs to answer the following questions:
* should a fee be charged to the patron?
* should the loan be immediately closed?
* should the loan be closed once the fee is paid? (maybe not needed yet)

And after that, charge the appropriate fee to the patron.

Have I missed something?

Reply:
Yes, correct. But a loan have to be closed when no cost defined at all. 

How would you suggest to model this case?

Reply:
> Yes, correct. But a loan have to be closed when no cost defined at all.
> 
> How would you suggest to model this case?

Ah, interesting. I understand your concern better.

If each fee is defined by a instance of a class, and each of the classes answered the question `should a fee be charged to this patron?`

Could this be answered when all of the fees answer that question negatively?

(In my mind, each of these item fees is really a strategy for charging patron fees, in that circumstance, a composite of zero fees could represent this situation)

Reply:
Thanks. When a fee is not present I return a NonChargeable fee object.